packages/mobile/src/components/details-tile/DetailsTile.tsx

```diff
@@ -148,10 +159,13 @@ export const DetailsTile = ({
   const isPlayingFullAccess = isPlaying && !isPreviewing
   const isUnpublishedScheduledRelease =
     track?.is_scheduled_release && track?.is_unlisted
-  const showPreviewButton =
-    isUSDCPurchaseGated &&
-    ((isOwner && !isCollection) || !hasStreamAccess) &&
-    onPressPreview
+
+  // Show play if user has access to the collection or any of its contents,
+  // otherwise show preview
+  const shouldShowPlay =
+    (isPlayable && hasStreamAccess) || doesUserHaveAccessToAnyTrack
+  const shouldShowPreview =
+    isUSDCPurchaseGated && !hasStreamAccess && !shouldShowPlay && onPressPreview
```

Comment:
doesn't this break the owner case for premium tracks where we want to show both?

Reply:
we're not showing both anymore! owner can go preview the track on the track page, but only show play for collections in all cases where the user has access

Reply:
Oh, is `DetailsTile` not used for the track page?

Reply:
I think it is, so we need a check here for collection vs track

Reply:
oh dang good catch